src/pages/Home.jsx

```diff
@@ -75,6 +75,8 @@ const Home = () => {
        * stored in the database and API call will fetch it.
        * So it should be safe to have the data eventually consistent.
        */
+      refetch()
```

Comment:
why do we need to refetch here? What does refetch do?

Reply:
When user create a new auction, user will be get "EVENT_TYPES.ItemListed" through event watcher of Web3.
Then to look a new auction on "Live Auction" page, it needs to get all auction list from the back-end server again.
So we need refetch function.

Reply:
But we are already doing that. This entire section adds the new auction to existing react-query cache: https://github.com/ryanprice/nft-market/blob/master/src/pages/Home.jsx#L82-L99

The comment section above also explains why we are doing it this way.

Reply:
As you can see the issue, the refetch doesn't work.
So I added the refetch function.

I will review the existing section in more detail.

Reply:
I reviewed the code section.

"queryClient.getQueryData" fetch only the data from a cache.
When user create a new "New Listing" and it is confirmed, Front-end should re-fetch a new data from server.
In other words, we can't get a new data from the cache.
Then we can use refetch() function.

Reply:
k I do see a bug in the handler. But as I mentioned above, we still shouldn't be refetching from the server whenever there's a new item listed event. That will overload the server with a lot of API calls. 

The data for the new item listed event is inside the handler as `data` variable.

The correct logic for when an item listed event is detected should be:
- check if there are cached data for item listed query
- If there are, append `data` to existing cache data. This will refresh the UI
- If there aren't, set `data` as the first page in the cached query data

Reply:
I have a question.
When user create a new "New Listing" and it is confirmed, Front-end should re-fetch a new data from server.
If we do as your logic, when do the refetch call?

Reply:
We have a watcher in the client: `setupWatcher.js`. This watcher is initialized at page load and it will listen to all the events our contract emits (including itemListed events). 

When an event is seen by the watcher, in the callback we use `Pubsub js` to publish to a bunch of subscribers. The handler you are trying to change now will get called because it is subscribing to the item listed event topic. The handler should then update the react-query cache data by appending the new data (triggered event) to the cache, and therefore updating the UI.

We don't need to refetch from server every time there is an event triggered. That's what our event watcher is for.

Reply:
The current code don't set any data to the cache of React Query (ex: EVENT_TYPES.ItemListed).
const currentData = queryClient.getQueryData()
'currentData' is always 'undefined'.

I think we should add a module that add the current data to the cache when the initial loading. Is that right?

Even if new data is added to the cache, UI doesn't update.
And refetch() is only called when 'EVENT_TYPES.ItemListed' event occurs.
It isn't overload.

Reply:
https://github.com/Cardinal-Entertainment/nft-market/pull/200 this PR has the correct solution. The bug was that we are not passing in chainId as the query key, so no data was appended to cache, which means the UI is not refreshed. 

And exactly. refetch is called when `EVENT_TYPES.ItemListed` occurs. And that can happen many times in a second when users start to listing items for auction. We don't need to send useless traffic to server when the event handler already have that data.

Reply:
That's good.
But when `EVENT_TYPES.ItemListed` occurs, how can we know whether that data exists or not?